scripts/idl/test_xml_parser.py

```diff
@@ -0,0 +1,197 @@
+#!/usr/bin/env python3
+# Copyright (c) 2022 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+import io
+
+from typing import Optional, Union, List
+
+try:
+    from idl.matter_idl_types import *
+    from idl.zapxml import ParseSource, ParseXmls
+except:
+    import os
+    import sys
+
+    sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))
+
+    from idl.matter_idl_types import *
+    from idl.zapxml import ParseSource, ParseXmls
+
+
+def XmlToIdl(what: Union[str, List[str]]) -> Idl:
+    if not isinstance(what, list):
+        what = [what]
+
+    sources = []
+    for idx, txt in enumerate(what):
+        sources.append(ParseSource(source=io.StringIO(txt), name=("Input %d" % (idx + 1))))
+
+    return ParseXmls(sources, include_meta_data=False)
+
+
+class TestXmlParser(unittest.TestCase):
+
+    def testEmptyInput(self):
+        idl = XmlToIdl('<configurator/>')
+        self.assertEqual(idl, Idl())
+
+    def testCluster(self):
+        idl = XmlToIdl('''<?xml version="1.0"?>
+            <!-- Comments should be supported -->
+            <configurator>
+              <cluster>
+                <name>Test Cluster</name>
+                <code>0x1234</code>
+
+                <attribute side="server" code="11" type="INT32U" min="0" max="2" isNullable="true" reportable="true" writable="false">SomeIntAttribute</attribute>
```

Comment:
Isn't the goal of these test snippets is to mimick XMLs that will be run through this parser so that we know it'll work when you throw real XMLs at it? And aren't those real XMLs coming from the existing XMLs checked into the tree, i.e ZAP XMLs?

Reply:
The goal for these tests is to validate that XML can be converted to 'data'.
In our case, we are ignoring some bits (like defines or unused tags like 'dmain' in clusters or 'description' or things like that).

I could add a test of 'it ignores things it does not care about' ... but not sure how valuable that is.

Reply:
The PR includes a CI test that we are ingesting zap xmls fully and without error.

Reply:
Also added a separate test to validate extra bits are not causing issues or getting read somewhere else in the Idl